weld-rendering/app/src/main/java/styler/weld/rendering/MainActivity.kt

```diff
@@ -1,56 +1,80 @@
 package styler.weld.rendering
 
 import android.os.Bundle
+import android.view.LayoutInflater
 import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.google.gson.reflect.TypeToken
 import kotlinx.android.synthetic.main.activity_main.*
 import styler.weld.rendering.adapter.WidgetAdapter
-import styler.weld.rendering.models.local.articlelist.ArticleListItem
+import styler.weld.rendering.factory.WidgetFactory
+import styler.weld.rendering.models.remote.WidgetData
 import styler.weld.rendering.models.remote.WidgetDefinitionList
 import styler.weld.rendering.utils.GsonUtils
-import styler.weld.rendering.viewholder.ArticleListViewHolder
-import styler.weld.rendering.viewholder.ItemListViewHolder
-import styler.weld.rendering.viewholder.WidgetFactory
+import styler.weld.rendering.viewholder.*
 import java.lang.reflect.Type
 
 
 class MainActivity : AppCompatActivity() {
 
-    private var widgetData: Widget? = null
+    private var widgetDataList = listOf<WidgetData>()
+
+    private val widgetFactory = WidgetFactory()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        initData()
         initViews()
     }
 
     private fun initViews() {
-        initData()
         setUpRecyclerView()
     }
 
     private fun setUpRecyclerView() {
-        val widgetFactory = WidgetFactory();
-        widgetFactory.register("article_list", ArticleListViewHolder::create )
-        widgetFactory.register("item_list", ItemListViewHolder::create)
+        widgetFactory.register("banner") { parent ->
+            val view = LayoutInflater.from(parent.context)
+                .inflate(R.layout.banner_row_item, parent, false)
+            return@register BannerViewHolder(view)
+        }
+
+        widgetFactory.register("item_list") { parent ->
+            val view = LayoutInflater.from(parent.context)
+                .inflate(R.layout.item_list_row_item, parent, false)
+            return@register ItemListViewHolder(view)
+        }
 
-        // examples in case you would handle area and feed in the same
-//        widgetFactory.register("area_selection", ....factory)
-//        widgetFactory.register("feed", ....feedfactory)
-//        widgetData.add(0, new WidgetData(type: "area_selection"))
-//        widgetData.add(widgetData.length - 1, new WidgetData(type: "feed"))
+        widgetFactory.register("shop_list") { parent ->
+            val view = LayoutInflater.from(parent.context)
+                .inflate(R.layout.shop_list_row_item, parent, false)
```

Comment:
this part here `R.layout.shop_list_row_item` should be owned by the ShopListViewHolder. this method here, that you do as a lamba, can you add it in or near the article list view holder?

I'm thinking I would expect something like:

```
widgetFactory.register("article_list") { parent -> ArticleListViewHolder.create(parent) }
```

Reply:
Makes sense. Fixed in https://github.com/ludugz/weld-rendering/pull/2/commits/81001fa5ebaac0aa34abbc7f2ef58feac559b485